### PR TITLE
VTN-17934 update code to use new ChunkResult message

### PR DESCRIPTION
**Pull request by seta-namnguyen**
Link to Ticket: https://steel-ventures.atlassian.net/browse/VTN-17934

Reviewers
- @matryer 
- @tnguyen-veri 

Purpose
-------
- update code to use new ChunkResult message

### CODE REVIEW

**Comment by matryer** [comment]
@seta-namnguyen How can we test that this is working?

**Comment by matryer** [comment]
@seta-namnguyen Do we want to proceed with this?

**Comment by seta-namnguyen** [comment]
> @seta-namnguyen How can we test that this is working?

For now, we can deploy to stage for testing

**Comment by matryer** [comment]
@seta-namnguyen ping

**Comment by seta-namnguyen** [comment]
@matryer,
ChunkResult message is ready in core. Please help me merge this pull request!
Thanks

**Comment by matryer** [comment]
@seta-namnguyen We just need to deploy one of the sample engines (I recommend the `http_proxy` one) to Stage to prove this is working, and then we can merge the PR. How does that sound?

File: engine/engine.go [hunk 1]
```diff
@@ -239,8 +239,7 @@ func (e *Engine) processMessageMediaChunk(ctx context.Context, msg *sarama.Consu
 		finalUpdateMessage.Status = chunkStatusIgnored
 		return nil
 	}
-	// send output message
-	outputMessage := mediaChunkMessage{
+	outputMessage := &mediaChunkMessage{
```
Comment (tnguyen-veri) [inline engine/engine.go]: It is confusing that this is typed "mediaChunkMessage" but it's actually an engine output, which are two different things in Edge.

File: engine/kafka.go [hunk 2]
```diff
@@ -141,6 +142,17 @@ type chunkProcessedStatus struct {
...
 
+type chunkResult struct {
+	Type         messageType        `json:"type,omitempty"`         // always messageTypeChunkResult
+	TimestampUTC int64              `json:"timestampUTC,omitempty"` // milliseconds since epoch
+	TaskID       string             `json:"taskId,omitempty"`       // Task ID the chunk belongs to
+	ChunkUUID    string             `json:"chunkUUID,omitempty"`    // UUID of chunk for which status is being reported
+	Status       chunkStatus        `json:"status,omitempty"`       // Processed status
+	ErrorMsg     string             `json:"errorMsg,omitempty"`     // Optional error message in case of ERROR status
+	InfoMsg      string             `json:"infoMsg,omitempty"`      // Optional message for anything engine wishes to report
+	EngineOutput *mediaChunkMessage `json:"engineOutput,omitempty"` // Engine output when chunk processed successfully, nil otherwise
```
Comment (tnguyen-veri) [inline engine/kafka.go]: Same as above

**Comment by tnguyen-veri** [comment]
Also agreed with Mat that need to verify change works, by testing with a sample engine, before merging to master.

**Comment by neovinhtru** [comment]
Hi @tnguyen-veri, @matryer 
I updated this engine send ChunkResult message to contain a status indicating the processing status of the chunk, either SUCCESS, ERROR or IGNORED (identical to current statuses)
https://steel-ventures.atlassian.net/browse/VTN-20317

I built and tested with Amazon Rekognition - Face Detection toolkit engine. It working as well
https://cms.stage.veritone.com/#/media-details/450998069

https://console.aws.amazon.com/cloudwatch/home?region=us-east-1#logEventViewer:group=/aws/ecs/stage-rt;stream=edge-coordinator-green/edge-coordinator-green-stage-rt/1e753169-0859-42ae-9324-64f2975b6e05;filter=result;start=2019-04-19T09:58:03Z

https://console.aws.amazon.com/cloudwatch/home?region=us-east-1#logEventViewer:group=/aws/ecs/stage-rt;stream=Fargate/EM-stage-rt-68d037d3-58b7-4adf-a5cb-7b249aa55007-E1024-AmazonRekognitionFaceDetectiontoolkit_F/b0cef840-bc06-445b-a50b-8c9f853a3c54

Can you help me review this PR?

Thanks!

**Pull request merged**